dotnet/sdk-phase-1.md

```diff
@@ -0,0 +1,1052 @@
+      https://github.com/temporalio/sdk-dotnet
+    * Disable GH pages
+    * Archive it
+    * Consider deletion at some point in the future
+  * Rename https://github.com/temporalio/sdk-dotnet-incubating to https://github.com/temporalio/sdk-dotnet
+  * Update DNS for https://dotnet.temporal.io/ to point to vercel-based API docs page
+
+Doing the repo rename was deemed easier than going through manually closing issues, closing PRs, creating empty `main`
+branch, and keeping `master` branch around for those reading older code.
+
+## Namespace/Package
+
+* Will publish as `Temporalio`
+  * Need to reserve prefix `Temporalio` on NuGet, see
+    https://learn.microsoft.com/en-us/nuget/nuget-org/id-prefix-reservation
+* Namespace will be `Temporal`
+  * See discussion topics later about this being `Temporal.Sdk` instead
+
+## Workflow Definition
+
+Here's a definition for an other-language-defined workflow (from Python README):
+
+```csharp
+namespace MyNamespace;
+
+using System.Text.Json.Serialization;
+using Temporal.Workflow;
+
+public record GreetingInfo
+{
+    [JsonPropertyName("salutation")]
+    public string Salutation { get; init; } = "Hello";
+
+    [JsonPropertyName("name")]
+    public string Name { get; init; } = "<unknown>";
+}
+
+[Workflow]
+public interface IGreetingWorkflow
+{
+    static readonly IGreetingWorkflow Ref = Refs<IGreetingWorkflow>.Instance;
+
+    [WorkflowRun]
+    Task<string> RunAsync(GreetingInfo info);
+
+    [WorkflowSignal]
+    Task UpdateInfoAsync(GreetingInfo info);
+
+    [WorkflowSignal]
+    Task CompleteWithGreetingAsync();
+
+    [WorkflowQuery]
+    string CurrentGreeting();
+}
+```
+
+Notes:
+
+* We use regular .NET JSON serialization (see data conversion later)
+* Workflow attributes, and in the future workflow runtime features, are in the `Temporal.Workflow` namespace
+* We only support workflow interfaces right now because we don't have a workflow implementation yet
+  * For now, everything non-static must have some form of `[Workflow` attribute, and not have a default implementation
+* Every workflow interface must have the non-inheritable `Workflow` attribute
+  * Optional `Name` positional property can be given. Default is the unqualified type name with the "I" prefix removed
+    if it's an interface (which it has to be currently) and the next letter is also capital
+    * So what does `IDWorkflow` become? Yes, `DWorkflow` :-( Weighed as an acceptably rare cost for obvious benefit
+* Every workflow must explicitly have one and only one method with a non-inheritable `WorkflowRun` attribute
+  * Can be in base class, but must also be overridden in this workflow class w/ the attribute even if it just delegates
+    to super
+  * Method must return a `Task`
+* Workflow signals are methods with the non-inheritable `WorkflowSignal` attribute
+  * Can be in base class, but if overridden, must also have the attribute
+  * Method must return a `Task` and can accept arguments
```

Comment:
> Is there a way to support both sync and async handlers?

Technically yes, but note that the presence of `Task` return does not require that the function be async. In fact you'll get a warning if you mark a `Task`-returning function as `async` if you don't `await` anywhere in it. But I understand it can be annoying to return a `Task` if you have no need for asynchrony in the signal.

After confirming as I have with activities that this won't cause type inference ambiguity I can support non-async signals I believe. I kinda liked the always-one-way myself.